python/res/enkf/jobs/enkf_scaling_job.py

```diff
@@ -0,0 +1,220 @@
+# -*- coding: utf-8 -*-
+
+import sys
+import configsuite
+
+from copy import deepcopy
+from collections import namedtuple
+
+from res.enkf.jobs import enkf_scaling_job_config
+from res.enkf.jobs.enkf_scaling_pca import get_scaling_factor
+from res.enkf import LocalObsdata, ActiveList
+from res.enkf.jobs.enkf_scaling_get_data import get_measured_data, get_scaled_matrix
+
+
+def main_entry_point(ert, user_config_dict):
+    """
+    Takes an instance of EnkFMain and a user config dict, will do some pre-processing on
+    the user config dict, set up a ConfigSuite instance and validate the job before control
+    is passed to the main job.
+    """
+
+    config_dict = _find_and_expand_wildcards(ert, user_config_dict)
+
+    config = setup_configuration(config_dict, enkf_scaling_job_config._build_schema())
+
+    if not valid_configuration(config):
+        sys.exit(1)
+
+    if not valid_job(ert.getObservations(), config):
+        sys.exit(1)
+
+    enkf_scale_job(ert, config)
+
+
+def enkf_scale_job(ert, config):
```

Comment:
Should become a class `ScaleJob` taking the observations (not `ert` if it does need it) and the `config`.